src/crypto/NoteEncryption.cpp

```diff
@@ -41,7 +41,7 @@ void KDF(unsigned char *K,
 }
 
 template<size_t MLEN>
-NoteEncryption<MLEN>::NoteEncryption(uint256 seed) : times_ran(0), seed(seed) {
+NoteEncryption<MLEN>::NoteEncryption(uint256 seed) : nonce(0), seed(seed) {
```

Comment:
I'm a little dubious about having this class be stateful and keeping track of a current nonce. I don't think it actually helps to make sure that nonces are distinct; we have to review the calling code to ensure it's being used correctly anyway.